src/Atlas.cc

```diff
@@ -315,12 +315,16 @@ void Atlas::PreSave()
             return elem1->GetId() < elem2->GetId();
         }
     };
-    std::copy(mspMaps.begin(), mspMaps.end(), std::back_inserter(mvpBackupMaps));
-    sort(mvpBackupMaps.begin(), mvpBackupMaps.end(), compFunctor());
+    {
+        lock_guard<mutex> lockCurr(mpCurrentMap->mMutexMapUpdate);
```

Comment:
Do you know what members mMutexMapUpdate guards? I would expect it to guard members of Map, not of Atlas.

Reply:
yeah this guards any changes to the maps themselves, so while we lock the atlas, its possible the [localMapping](https://github.com/viamrobotics/ORB_SLAM3/blob/master/src/LocalMapping.cc#L1280) and LoopClosing threads are still looking to use the maps, so we want to make sure those threads are finished making any changes before we save the map. Locking the atlas should be enough probably since the other threads depend on it, but was not sure if there would be a weird corner case where the other threads already accessed the map before we locked the atlas

Reply:
I'm confused why we lock the current map before copying mspMaps into mvpBackupMaps. How are mspMaps and mvpBackupMaps related to the current map? Also is this performing a deep copy? If not, the maps could continue to be modified after copying.

In general, I'm worried about locking a mutex for a different class, since mutexes are general class-private, which helps avoid defining a global mutex ordering in order to prevent deadlocks. Did this fix a bug? If not, I wouldn't recommend making this change.